widgets/blueprints/src/TerraformActions.ts

```diff
@@ -7,7 +7,15 @@ export default class TerraformActions {
         return this.toolbox.getInternal().doPost('/terraform/blueprint', { body });
     }
 
-    doGetTemplateModules(templateZipUrl: string) {
-        return this.toolbox.getInternal().doPost('/terraform/resources', { params: { zipUrl: templateZipUrl } });
+    doGetTemplateModules(templateZipUrl: string, username?: string, password?: string) {
+        let headers;
+        if (username) {
+            headers = { Authorization: `Basic ${btoa(`${username}:${password}`)}` };
+        }
```

Comment:
```suggestion
        const headers = username ?
            { Authorization: `Basic ${btoa(`${username}:${password}`)}` } :
            undefined;
```

What would you think about the suggestion above?
Just wanted to comment on that, so that we wouldn't have a situation when someone would override the `let` variable, when it shouldn't be necessary 😉 
(currently the `headers` variable is being assigned only 2 values. Because of that I thought that we could reassign it as a `const` 😉 )

Please let me know what you think

Reply:
> Just wanted to comment on that, so that we wouldn't have a situation when someone would override the let variable, when it shouldn't be necessary 😉

There is nothing easier than changing `const` to `let` in case anybody wanted to reassign the variable, I don't think `const` actually protects from anything. It's usage is enforced by eslint rule in all cases where it can be used in place of `let`, so in terms of understanding the code it doesn't have the semantics of "don't reassign".
But even if it did, why do you think we should try to guard against reassigning? I think it's impossible to say if it would be (or not) a desirable thing without knowing what the future change would be.

Reply:
You may be right about it
But still I think that there exists problem related to using `let` and assigning a single value to it ... just in another block of code 😁 
I think that this operation may be confusing, because why should we make seperate block lines just to assign one value to a variable?

Currently the situation is not as confusing, because we keep the lines underneath each other
But still I think that the question "where is the value of that `headers`?" may be unnecessarily being asked by a developer to himself

Reply:
But I think that's not the "deal breaker", so if you'd find that modification unnecessary, we may skip it
Just let me know 😉

Reply:
> I think that this operation may be confusing, because why should we make seperate block lines just to assign one value to a variable?

Yes, this is a good point after all. Updated.

Reply:
I see, that you've added changes related to this issue (thanks for addressing it! 💪 )
Because of that, I'm resolving this discussion